core/io/file_access_encrypted.cpp

```diff
@@ -236,8 +240,13 @@ bool FileAccessEncrypted::store_buffer(const uint8_t *p_src, uint64_t p_length)
 		ERR_FAIL_COND_V(data.resize(pos + p_length) != OK, false);
 	}
 
-	memcpy(data.ptrw() + pos, p_src, p_length);
-	pos += p_length;
+	if (p_length) {
```

Comment:
Not sure what can be `nullptr` in this case, `data.ptrw()` should never be `nullptr`, and `p_src` is checked before.

Reply:
The check above is: `ERR_FAIL_COND_V(!p_src && p_length > 0, false);` 

"If not p_src *and* p_length >0" this means that if p_src == nullptr we still hit the memcpy, but only if p_length == 0. Calling memcpy with a null dest or src causes UB, even with a length of 0.

Reply:
Seems like `data.ptrw()` is not guaranteed to not be `nullptr`, but maybe something like woult be better:
```c++
void *ptr = data.ptrw();
if (ptr) {
	memcpy(data.ptrw() + pos, p_src, p_length);
}
```

Reply:
> If not p_src and p_length > 0

Ah, indeed, I missed `p_length` part. But still, something like `if (ptr && p_src)` might be cleaner than checking size.

Reply:
Neither data nor p_src are guaranteed to not be nullptr. if p_length ==0 it is likely because p_src is null, which is a case this code explicitly supports.

if p_length == 0 and pos == 0 then BOTH p_src and data.ptrw() will be nullptr.

Reply:
At least in C, this situation now is well defined in latest standard, https://developers.redhat.com/articles/2024/12/11/making-memcpynull-null-0-well-defined#

Reply:
We currently use c++17 where it is not, this pattern leads to actual non-theoretical crashes right now, see https://github.com/godotengine/godot/pull/100422 for instance.